DatadogInternal/Sources/NetworkInstrumentation/NetworkInstrumentationFeature.swift

```diff
@@ -78,17 +164,23 @@ extension NetworkInstrumentationFeature {
     ///   - task: The created task.
     ///   - additionalFirstPartyHosts: Extra hosts to consider in the interception.
     func intercept(task: URLSessionTask, additionalFirstPartyHosts: FirstPartyHosts?) {
-        guard let request = task.originalRequest else {
+        guard let originalRequest = task.originalRequest else {
             return
         }
 
-        queue.async {
+        // async update to task leads to race condition where the currentRequest could already be sent to the transport
+        queue.sync {
```

Comment:
/question We could get rid of the `queue` and use locks only, no?

Reply:
lock has been removed.